user-integration-guide/symfony-integration.md

```diff
@@ -0,0 +1,218 @@
+
+## Step 4 - Bundle Zipkin and  Jaeger into the Application
+
+To visualize traces from our application, we have to bundle open source tracing tools [Zipkin](https://zipkin.io/) and [Jaeger](https://www.jaegertracing.io/) into our application using docker. 
+
+Let's add a `docker-compose.yaml` file in the root of our project with the content as follows:
+
+```
+version: '3.7'
+services:
+    zipkin:
+        image: openzipkin/zipkin-slim
+        ports:
+            - 9411:9411
+    jaeger:
+        image: jaegertracing/all-in-one
+        environment:
+            COLLECTOR_ZIPKIN_HTTP_PORT: 9412
+        ports:
+            - 9412:9412
+            - 16686:16686
+```
+
+To confirm that docker is installed and running on our system, we can run the hello world docker example using the command `docker run -it --rm hello-world`. If everything works well, run  `docker-compose up -d` to pull in Zipkin and Jaeger. This might take some time, depending on your internet connection speed.
+
+We can confirm that Zipkin is up by navigating to `http://localhost:9411/` on our browser. For Jaeger, navigating to `http://localhost:16686/` on our browser should display the Jaeger home page.
+
+![image](https://user-images.githubusercontent.com/22311928/110503699-bfa04e00-80fc-11eb-9186-c9b295d100f4.png)
+
+![image](https://user-images.githubusercontent.com/22311928/110504108-1f96f480-80fd-11eb-9a2b-a7b4faf8b11c.png)
+
+
+Now it is time to utilize our OpenTelemetry PHP Library to export traces to both Zipkin and Jaeger.
+
+## Step 5
+
+The entry point for all Symfony applications is the `index.php` file located in the `public` folder. Let's navigate to `public\index.php` to see what is happening. It is worthy of note that resources(namespaces, classes, variables) created within the `index.php` file are available within the entire application, by default the index file imports all auto loaded classes within the vendor folder. It also imports contents of the `.env` file. The other parts of the `index.php` file enable debugging as well as support request and response resolution using the application kernel. 
+
+To use open-telemetry specific classes we have to import them at the top of our index file, using the `use` keyword. This is what our imports look like:
+
+```
+use App\Kernel;
+use OpenTelemetry\Contrib\Jaeger\Exporter as JaegerExporter;
+use OpenTelemetry\Contrib\Zipkin\Exporter as ZipkinExporter;
+use OpenTelemetry\Sdk\Trace\Clock;
+use OpenTelemetry\Sdk\Trace\Sampler\AlwaysOnSampler;
+use OpenTelemetry\Sdk\Trace\SamplingResult;
+use OpenTelemetry\Sdk\Trace\SpanProcessor\BatchSpanProcessor;
+use OpenTelemetry\Sdk\Trace\TracerProvider;
+use OpenTelemetry\Trace as API;
+use Symfony\Component\Dotenv\Dotenv;
+use Symfony\Component\ErrorHandler\Debug;
+use Symfony\Component\HttpFoundation\Request;
+```
+
+Next, we create a sample recording trace using the `AlwaysOnSampler`,  class, just before the Kernel instance is created like below:
```

Comment:
You may want to reference the repo's implementation of the `AlwaysOnSampler` here, because it's subject to change.